crates/stages/src/stage.rs

```diff
@@ -77,30 +77,46 @@ impl ExecInput {
         &self,
         provider: &DatabaseProviderRW<'_, DB>,
         tx_threshold: u64,
-    ) -> Result<(RangeInclusive<TxNumber>, RangeInclusive<BlockNumber>, bool), StageError> {
+    ) -> Result<(Range<TxNumber>, RangeInclusive<BlockNumber>, bool), StageError> {
```

Comment:
we discussed before but should we consider going to `Range` instead of `RangeInclusive` everywhere to be consistent? recall you didnt love it, also recall me and @onbjerg not loving it at the time..no storng opinions just askign the q

Reply:
As I mostly fixed a lot of bugs with this, imo i feel like `RangeInclusive` for blocks is less error prone and same for `Range` for transactions. This was a reason why i used it as is, but in the end it is not a big of a deal having it one or another, and i wouldn't mind a lot if it gets changed.

Reply:
I think most confusion stems not from `Range` `RangeInclusive` but from edge cases when the block does not have transactions but it does have an index on the next block transaction. 
 https://github.com/paradigmxyz/reth/blob/6cbb639f6b4f32ee9908007b64b0a61ea50879d9/crates/storage/db/src/tables/models/blocks.rs#L35-L50